test: Re-enable ChromeOS testing

avelad opened the pull request.

joeyparrish commented:
I don't think our lab ChromeOS device is alive right now.

avelad commented:
> I don't think our lab ChromeOS device is alive right now.

Can you start it so we can adjust the tests and add it to the normal cycle? Thanks!

joeyparrish commented:
I will work on that later today if I can.  I need to go into the lab, update the device, and check that [ChromeOS WebDriver Server](https://github.com/shaka-project/generic-webdriver-server/tree/main/backends/chromeos#readme) is still functional.  We haven't used it in a long time, so it may not be any more.

When the device is on and updated, and WebDriver Server is working, I need to add it to the [Shaka Lab Node](https://github.com/shaka-project/shaka-lab/tree/main/shaka-lab-node#readme) config.  Then it should be available to CI again.

avelad commented:
@shaka-bot test

joeyparrish commented:
I had already updated and fixed the ChromeOS device in our lab.  However, some time later, it seems to have rebooted or auto-updated and in the process, SSH was disabled.

The setup process is meant to keep SSH enabled across updates and reboots, but it's not working and needs to be fixed.

avelad commented:
> I had already updated and fixed the ChromeOS device in our lab. However, some time later, it seems to have rebooted or auto-updated and in the process, SSH was disabled.
> 
> The setup process is meant to keep SSH enabled across updates and reboots, but it's not working and needs to be fixed.

Only for ChromeOS or for all the lab?

joeyparrish commented:
I was referring to the setup process for ChromeOS in the lab.  After some more digging, I discovered that I was misremembering.

Before Generic WebDriver Server, we had a much more complicated system where we explicitly disabled auto-updates, and had to periodically run a custom tool that would check for updates, install those updates, then modify them to enable SSH and a read-write root partition before rebooting.  And it failed a lot.

[The docs for Generic WebDriver Server say](https://github.com/shaka-project/generic-webdriver-server/blob/main/backends/chromeos/device-setup.md#updating-chromeos):

> Updating ChromeOS may undo the "Enabling SSH" steps above, and they may need to be repeated periodically if auto-updates are enabled on the device.

So I'm working on a solution to that.  I have an idea that may work, but I haven't been able to test it yet.  There's a feature in the OS now called "DLC" that may allow us to install an extra piece of software that survives updates and is reapplied after each one.  If I can use that mechanism to enable SSH, and incorporate that into the setup instructions, ChromeOS WebDriver Server, which uses SSH to control the device, should continue to function across updates.

avelad commented:
@joeyparrish any update?

avelad commented:
Any eta @joeyparrish ? Should I close it?

joeyparrish commented:
> [The docs for Generic WebDriver Server say](https://github.com/shaka-project/generic-webdriver-server/blob/main/backends/chromeos/device-setup.md#updating-chromeos):
> 
> > Updating ChromeOS may undo the "Enabling SSH" steps above, and they may need to be repeated periodically if auto-updates are enabled on the device.
> 
> So I'm working on a solution to that. I have an idea that may work, but I haven't been able to test it yet.

It didn't work.  I don't have any new ideas at the moment.
